### PR TITLE
[DuckPlayer] 23. Ship Review Fixes

### PR DESCRIPTION
**Pull request by afterxleep**
Task/Issue URL: https://app.asana.com/0/1204099484721401/1208094354951735/f

**Description**:
Fixes issues found in DuckPlayer ship review
- Opens videos in Youtube app if present and user selects "Open in Youtube"
- Updates bookmarks images
- Updates history

**Steps to test this PR**:
1. Make yourself internal user
2. Add this custom config https://www.jsonblob.com/api/1276215786342309888 
3. Add a bookmark and a favorite while watching a video in DuckPlayer
4. Confirm custom icons are used

----

1. Disable Duck Player
2. Go to use-devtesting18.duckduckgo.com and search for a video
3. Tap "Watch in Youtube"
4. In the simulator of if Youtube app is not installed, video should load in DDG Browser
5. If Youtube app is installed, video should load in YT app

---

1. Set DuckPlayer to 'always ask'
2. Go to Youtube
3. Search for a video
4. Tap "Watch in Duck Player" (Dont enable remember choice)
5. Go back
6. Confirm you're taken to the video page and not results view.

### CODE REVIEW

File: DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift [hunk 1]
```diff
@@ -165,22 +198,28 @@ extension DuckPlayerNavigationHandler: DuckNavigationHandling {
...
-                if let videoParameterItem = queryItems.first(where: { $0.name == Constants.watchInYoutubeVideoParameter }),
-                   let id = videoParameterItem.value,
-                    let newURL = URL.youtube(id, timestamp: nil).addingWatchInYoutubeQueryParameter() {
-                        // These links should always skip the overlay
-                        duckPlayer.settings.allowFirstVideo = true
-                        Pixel.fire(pixel: Pixel.Event.duckPlayerWatchOnYoutube)
-                        webView.load(URLRequest(url: newURL))
-                        return
-                }
+        if url.scheme == "duck",
```
Comment (Bunn) [inline DuckDuckGo/DuckPlayer/DuckPlayerNavigationHandler.swift]: Maybe compare it to `NavigationalScheme.duck.rawValue` ?

File: DuckDuckGo/DuckPlayer/YoutubeOverlayUserScript.swift [hunk 2]
```diff
@@ -88,6 +88,7 @@ final class YoutubeOverlayUserScript: NSObject, Subfeature {
     
     let messageOriginPolicy: MessageOriginPolicy = .only(rules: [
         .exact(hostname: "sosbourne.duckduckgo.com"),
+        .exact(hostname: "use-devtesting18.duckduckgo.com"),
```
Comment (Bunn) [inline DuckDuckGo/DuckPlayer/YoutubeOverlayUserScript.swift:91]: Just to confirm, this will be removed once the web is deployed, correct?

**Review by Bunn — APPROVED** [review]
LGTM. Small comments in the code.

Only thing I'm not sure if it's working as expected:
* Save DuckPlayer video as favorite
* Disable DuckPlayer
* Open the saved video in favorites
My expectation would be to open the youtube video in youtube
The actual is a blank loading page that never finishes loading

File: DuckDuckGo/DuckPlayer/YoutubeOverlayUserScript.swift (hunk 2 above)
Comment (afterxleep) [reply DuckDuckGo/DuckPlayer/YoutubeOverlayUserScript.swift:91]: Yes, both those hosts will be gone.

**Comment by afterxleep** [comment]
> My expectation would be to open the youtube video in youtube
The actual is a blank loading page that never finishes loading

This is a known issue that will be fixed in a subsequent PR

**Pull request merged**